collector/shards_response.go

```diff
@@ -0,0 +1,20 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package collector
+
+// ShardResponse is a representation of an Elasticsearch Shard
```

Comment:
Because this is such a simple struct, I would put it in the `collector/shards.go` file. There's not much benefit to splitting it out. I do realize that has previously been the convention. It wouldn't block merging for me, but since there are other changes to make, I think it's better to clean it all up at the same time.